tradeexecutor/ethereum/one_delta/one_delta_valuation.py

```diff
@@ -0,0 +1,48 @@
+"""Value model for 1delta trade based on Uniswap v3 market price.
+"""
+import datetime
+
+from tradeexecutor.ethereum.one_delta.one_delta_live_pricing import OneDeltaLivePricing
+from tradeexecutor.state.position import TradingPosition
+from tradeexecutor.state.types import USDollarAmount
+from tradeexecutor.ethereum.eth_valuation import EthereumPoolRevaluator
+
+
+class OneDeltaPoolRevaluator(EthereumPoolRevaluator):
+    """Re-value assets based on their on-chain price.
+
+    Does directly JSON-RPC call to get the latest price in the Uniswap pools.
+
+    Only uses direct route - mostly useful for testing, may not give a realistic price in real
+    world with multiple order routing options.
+
+    .. warning ::
+
+        This valuation method always uses the latest price. It
+        cannot be used for backtesting.
+    """
+
+    def __init__(self, pricing_model: OneDeltaLivePricing):
+        assert isinstance(pricing_model, OneDeltaLivePricing)
+        super().__init__(pricing_model)
+
+    def __call__(
+        self,
+        ts: datetime.datetime,
+        position: TradingPosition,
+    ) -> tuple[datetime.datetime, USDollarAmount]:
+        pair = position.pair.get_pricing_pair()
+
+        quantity = position.get_quantity()
+        
+        # Cannot do pricing for zero quantity
+        if quantity == 0:
+            return ts, 0.0
+
+        price_structure = self.pricing_model.get_sell_price(ts, pair, quantity)
```

Comment:
This was originally meant to value position by "selling spot". Does this valuation work for OneDelta?

Reply:
Yes kinda, underlying it's still using Uniswap (v3 for our case for now) for buy and sell so the spot price should be more or less correct. 
There could be difference in the fee.
We can create another ticket to take 1delta quoter into use instead of current uniswap quoter.